make/config.go

```diff
@@ -37,35 +46,102 @@ func extractConfigError(path, input string, inErr error) error {
 	return err
 }
 
+func findConfigFile() (string, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	for {
+		fp := filepath.Join(dir, "ron.yaml")
+		if _, err := os.Stat(fp); err == nil {
+			return fp, nil
+		} else if !os.IsNotExist(err) {
+			return "", err
+		}
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			return "", nil
+		}
+		dir = parentDir
+	}
+}
+
+// LoadConfigFiles loads the default and override config files and returns
+// them as a slice. If defaultYamlPath is an empty string, the defaults
+// compiled into ron will be used instead. If overrideYamlPath is blank,
+// it will find the nearest parent folder containing a ron.yaml file and use
+// that file instead. In that case, the path to that file will be returned
+// so that the caller can change the working directory to that folder before
+// running further commands.
+func LoadConfigFiles(defaultYamlPath, overrideYamlPath string) ([]Config, string, error) {
+	configs := []Config{}
```

Comment:
It just didn't seem necessary since the structs are generally lightweight and we only have a couple of them. I will change them to pointers.